apps/example-server/src/main/case/case.routes.ts

```diff
@@ -12,8 +12,46 @@ const secret = process.env.ACCESS_TOKEN_SECRET ?? ''
 
 const authMiddleware = jwt({ secret })
 
-// GET /cases - list all
-app.get('/cases', authMiddleware, async (c) => {
+const checkCaseOwnershipMiddleware = async (ctx: Context, next: Next) => {
+    const payload = await ctx.get('jwtPayload')
+    if (!payload) {
+        return ctx.json(
+            { error: 'Unauthorized', message: 'Not authenticated' },
+            403,
+        )
+    }
+
+    const caseId = parseInt(ctx.req.param('id'), 10)
+    if (isNaN(caseId)) {
+        return ctx.json(
+            { error: 'Invalid ID', message: 'Case ID must be a number' },
+            400,
+        )
+    }
```

Comment:
This check can also be refactored into a global middleware. Or better yet, we can use `zValidator` for this:

```
// schema file
export const zIdParam = z.object({
  id: z.number(), // TODO: add auto transformation and custom error message
})

// in handler
app.post('/:id', zValidator('params', zIdParam), ...)
```